secrets/secrets.go

```diff
@@ -298,3 +322,35 @@ func NewSecrets(r io.Reader) (*Secrets, error) {
 	}
 	return secrets, nil
 }
+
+func csiPathParser(dir fs.FS) (Document, error) {
+	secretsDocument := Document{
+		Secrets: make(map[string]GenericSecret),
+	}
+	err := fs.WalkDir(
+		dir,
+		".",
+		func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+			if d.IsDir() {
+				return nil
+			}
+			// parse file
+			file, err := fs.ReadFile(dir, path)
```

Comment:
read the file fully into memory then use `bytes.NewReader` is very inefficient. just use `d.Open`, defer close after error check, would be much better.